src/BackgroundProcess/AjaxBackgroundProcess/AbstractAjaxBackgroundProcess.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+namespace WP_Statistics\BackgroundProcess\AjaxBackgroundProcess;
+
+use WP_Statistics\BackgroundProcess\AjaxBackgroundProcess\Jobs\VisitorColumnsMigrator;
+use WP_STATISTICS\Option;
+use WP_STATISTICS\User;
+use WP_Statistics\Utils\Request;
+
+/**
+ * Base class for handling background database migrations in an asynchronous manner.
+ *
+ * This class provides the core structure for running database migrations in the background 
+ * via AJAX requests. It ensures that migrations are executed in a controlled manner, preventing 
+ * duplication, tracking progress, and marking migrations as completed when finished.
+ *
+ * It maintains a list of registered migrations, handles the retrieval of pending tasks, 
+ * and ensures proper state tracking to resume operations in case of interruptions.
+ */
+abstract class AbstractAjaxBackgroundProcess
+{
+    /**
+     * Number of records processed per batch.
+     *
+     * @var int
+     */
+    protected $batchSize = 50;
+
+    /**
+     * Total number of records to migrate.
+     *
+     * @var int
+     */
+    protected $total = 0;
+
+    /**
+     * Number of records already migrated.
+     *
+     * @var int
+     */
+    protected $done = 0;
+
+    /**
+     * Number of records remaining to be migrated.
+     *
+     * @var int
+     */
+    protected $remains = 0;
+
+    /**
+     * Holds the current migration class name.
+     *
+     * @var string|null
+     */
+    protected static $currentMigration;
+
+    /**
+     * List of available migrations.
+     *
+     * @var array
+     */
+    public static $migrations = [
+        'visitor_columns_migrate' => VisitorColumnsMigrator::class,
+    ];
```

Comment:
I really not sure it would be a good idea to register the migrations in the abstract or not.

Anyway, let's see how it goes.